packages/grid/x-data-grid-pro/src/components/headerFiltering/GridHeaderFilterAdornment.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import {
+  GridFilterItem,
+  GridFilterOperator,
+  useGridApiContext,
+  GridColDef,
+} from '@mui/x-data-grid';
+import { unstable_useId as useId } from '@mui/utils';
+import { unstable_gridHeaderFilteringMenuSelector } from '@mui/x-data-grid/internals';
+import { useGridRootProps } from '../../hooks/utils/useGridRootProps';
+import { OPERATOR_SYMBOL_MAPPING } from './constants';
+
+function GridHeaderFilterAdornment(props: {
+  operators: GridFilterOperator<any, any, any>[];
+  field: GridColDef['field'];
+  item: GridFilterItem;
+  applyFilterChanges: (item: GridFilterItem) => void;
+  headerFilterMenuRef: React.MutableRefObject<HTMLButtonElement | null>;
+  buttonRef: React.Ref<HTMLButtonElement>;
+}) {
+  const { operators, item, field, buttonRef, headerFilterMenuRef, ...others } = props;
+
+  const buttonId = useId();
+  const menuId = useId();
+
+  const rootProps = useGridRootProps();
+  const apiRef = useGridApiContext();
+  const open = Boolean(
+    unstable_gridHeaderFilteringMenuSelector(apiRef) === field && headerFilterMenuRef.current,
+  );
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    headerFilterMenuRef.current = event.currentTarget;
+    apiRef.current.showHeaderFilterMenu(field);
+  };
+
+  if (!rootProps.slots.headerFilterMenu) {
+    return null;
+  }
+
+  return (
+    <React.Fragment>
+      <rootProps.slots.baseInputAdornment position="start">
+        <rootProps.slots.baseIconButton
+          id={buttonId}
+          ref={buttonRef}
+          aria-label={apiRef.current.getLocaleText('filterPanelOperator')}
+          title={apiRef.current.getLocaleText('filterPanelOperator')}
+          aria-controls={menuId}
+          aria-expanded={open ? 'true' : undefined}
+          aria-haspopup="true"
+          tabIndex={-1}
+          size="small"
+          onClick={handleClick}
+          sx={{
+            width: 22,
+            height: 22,
+            padding: '0px 0px 2px 2px',
+          }}
```

Comment:
The `sx` value could be extracted, it would avoid continuously re-allocating it.

Reply:
Nice recommendation, applied 👍